apps/src/code-studio/sectionsRedux.js

```diff
@@ -0,0 +1,49 @@
+import Immutable from 'immutable';
+import _ from 'lodash';
+
+// Action types
+export const SET_SECTIONS = 'sections/SET_SECTIONS';
+export const SELECT_SECTION = 'sections/SELECT_SECTION';
+
+// Action Creators
+export const setSections = sections => ({
+  type: SET_SECTIONS,
+  sections
+});
+
+export const selectSection = sectionId => ({
+  type: SELECT_SECTION,
+  sectionId
+});
+
+const SectionData = Immutable.Record({
+  selectedSection: null,
+  sectionsLoaded: false,
+  sectionIds: [],
+  bySection: {}
```

Comment:
Since redux pushes us toward having a single source of truth for state on a page, it makes sense to me that'd you want to remove `sectionIds` and depend only on `bySection`.  Is order important?  If so, maybe `bySection` could be an [`OrderedMap`](https://facebook.github.io/immutable-js/docs/#/OrderedMap)?

I'm in favor of simplifying `bySection` and renaming it to something like `nameById` if that's all you're using it for right now.